forge-gui/res/cardsfolder/a/armed_and_armored.txt

```diff
@@ -2,12 +2,9 @@ Name:Armed and Armored
 ManaCost:1 W
 Types:Instant
 A:SP$ AnimateAll | Cost$ 1 W | Types$ Creature,Artifact | ValidCards$ Vehicle.YouCtrl | SubAbility$ ChooseDwarf | StackDescription$ Vehicles {p:You} controls become artifact creatures until end of turn. | SpellDescription$ Vehicles you control become artifact creatures until end of turn. Choose a Dwarf you control. Attach any number of Equipment you control to it.
-SVar:ChooseDwarf:DB$ ChooseCard | Defined$ You | Mandatory$ True | Choices$ Dwarf.YouCtrl | ChoiceTitle$ Choose a Dwarf you control | StackDescription$ {p:You} chooses a Dwarf they control and attaches any number of Equipment they control to it. | ImprintChosen$ True | SubAbility$ ChooseEquipment
-SVar:ChooseEquipment:DB$ ChooseCard | Defined$ You | MinAmount$ 0 | Amount$ X | Choices$ Equipment.YouCtrl | StackDescription$ None | ChoiceTitle$ Choose any number of Equipment you control | ForgetChosen$ True | SubAbility$ DeployDwarf
-SVar:DeployDwarf:DB$ RepeatEach | RepeatSubAbility$ ArmDwarf | RepeatCards$ Card.ChosenCard | SubAbility$ DBCleanup
-SVar:ArmDwarf:DB$ Attach | Object$ Remembered | Defined$ Imprinted | StackDescription$ None
-SVar:DBCleanup:DB$ Cleanup | ClearChosenCard$ True | ClearRemembered$ True | ClearImprinted$ True
-SVar:X:Count$Valid Equipment.YouCtrl
+SVar:ChooseDwarf:DB$ ChooseCard | Defined$ You | Mandatory$ True | Choices$ Dwarf.YouCtrl | ChoiceTitle$ Choose a Dwarf you control | StackDescription$ {p:You} chooses a Dwarf they control and attaches any number of Equipment they control to it. | SubAbility$ ArmDwarf
+SVar:ArmDwarf:DB$ Attach | Object$ Equipment.YouCtrl | Defined$ ChosenCard | Optional$ True | StackDescription$ None | SubAbility$ DBCleanup
```

Comment:
you can do `Object$ Equipment.YouCtrl | Choices$ Dwarf.YouCtrl`

Reply:
I don't think I can
That combination seems to be meant for another path here:
https://github.com/Card-Forge/forge/blob/e2b7cd2f5fc46021599dc18369ad44580abf42fe/forge-game/src/main/java/forge/game/ability/effects/AttachEffect.java#L103

So I can't choose cards if not all equipment could attach, that should be wrong for this case

Reply:
probably add Flag for Optional

Reply:
Such change would break _Gift of Doom_:
> The chosen recipient must be able to legally be enchanted by the Aura, so a creature with protection from black can’t be chosen this way.

Reply:
Idea: check if the attachment list is > 1 AND Optional, skip the step

i don't find it right now, there is another one that can attach multiple Equipment at once where this matters, but i don't think it was Optional